src/Transport/GrpcTransport.php

```diff
@@ -283,4 +307,66 @@ private static function loadClientCertSource(callable $clientCertSource)
     {
         return call_user_func($clientCertSource);
     }
+
+    /**
+     * @param Call $call
+     * @param array $headers
+     *
+     * @return void
+     */
+    private function logRequest(Call $call, array $headers)
+    {
+        $logger = $this->logger;
+        $timestamp = date(DATE_RFC3339);
+
+        $debugEvent = [
+            'timestamp' => $timestamp,
+            'severity' => 'DEBUG',
+            'jsonPayload' => [
+                'request.rpcName' => $call->getMethod(),
+                'request.headers' => $headers,
+                'request.payload' => $call->getMessage()
+            ]
+        ];
+
+        $logger->debug(json_encode($debugEvent));
+    }
+
+    /**
+     * @param mixed $response
+     * @param mixed $status
+     *
+     * @return void
+     */
+    private function logResponse(mixed $response, mixed $status)
+    {
+        $logger = $this->logger;
```

Comment:
Why do it this way ?

Reply:
No specific reason. This can be easily changed to `$this->logger-><level>()` below.

Reply:
Seems strange to me to set a local variable that we don't need